opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -211,14 +256,36 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
             );
         }
 
-        let (res_sender, res_receiver) = oneshot::channel();
-        self.message_sender
-            .try_send(BatchMessage::Shutdown(res_sender))
+        let (sender, receiver) = mpsc::sync_channel(1);
```

Comment:
I remember we discussed replacing oneshot with mpsc::sync_channel to remove dependency on external crate. However, I think oneshot is still lightweight option for point-to-point communication. Not something for this PR, something to evaluate again for future.

Reply:
Yes, I think the issue is oneshot channel doesn't provide recv with timeout. Let's revisit it later as the assumption could change.